src/nodes.coffee

```diff
@@ -2635,7 +2637,8 @@ exports.Op = class Op extends Base
     if op is 'do'
       return Op::generateDo first
     if op is 'new'
-      return first.newInstance() if first instanceof Call and not first.do and not first.isNew
+      if (firstCall = first.unwrap()) instanceof Call and not firstCall.do and not firstCall.isNew
```

Comment:
this and the two other below changes to `nodes.coffee` are places where a `Call` that's now wrapped in a `Value` (due to the grammar changes) needed to be unwrapped in order for tests to pass